client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,31 +1,183 @@
+					onClick={ onAnnualPlanClick }
+				>
+					{ translate( 'Pay annually' ) }
+				</Button>
+			</ButtonGroup>
+		</div>
+	);
+};
+
+interface PlanPriceOfferProps {
+	plan?: Plan;
+	currencyCode?: string;
+	originalMonthlyPrice?: number;
+	introOfferMonthlyPrice?: number;
+	originalFullPrice?: number;
+	introOfferFullPrice?: number;
+	introOfferEnabled: boolean;
+	isIntroOfferComplete: boolean;
+}
+
+const PlanPriceOffer = ( props: PlanPriceOfferProps ) => {
+	const translate = useTranslate();
+
+	const {
+		plan,
+		originalFullPrice,
+		introOfferFullPrice,
+		introOfferEnabled,
+		isIntroOfferComplete,
+		introOfferMonthlyPrice,
+		originalMonthlyPrice,
+		currencyCode,
+	} = props;
+
+	const showIntroOffer =
+		introOfferEnabled &&
+		! isIntroOfferComplete &&
+		introOfferMonthlyPrice &&
+		originalMonthlyPrice &&
+		introOfferFullPrice &&
+		originalFullPrice &&
+		currencyCode;
+
+	if ( ! showIntroOffer ) {
+		return (
+			<div className="import__upgrade-plan-price">
+				<PlanPrice rawPrice={ originalMonthlyPrice } currencyCode={ currencyCode } isSmallestUnit />
+				<span className="plan-time-frame">
+					<small>{ plan?.getBillingTimeFrame() }</small>
+				</span>
+			</div>
+		);
+	}
+
+	const billingTimeFrame = translate(
+		'per month, %(discountedPrice)s billed annually for the first year, %(originalPrice)s afterwards, excl. taxes',
+		{
+			args: {
+				discountedPrice: formatCurrency( introOfferFullPrice, currencyCode, { stripZeros: true } ),
+				originalPrice: formatCurrency( originalFullPrice, currencyCode, {
+					isSmallestUnit: true,
+					stripZeros: true,
+				} ),
+			},
+			comment:
+				'excl. taxes stands for excluding taxes; discountedPrice is a formatted price like $150; originalPrice is a formatted price like $200',
+		}
+	);
+
+	return (
+		<div className="import__upgrade-plan-price">
+			<Badge type="info-purple" className="import__upgrade-plan-price-badge">
+				{ translate( 'One time offer' ) }
+			</Badge>
+			<div className="import__upgrade-plan-price-group">
+				<PlanPrice
+					rawPrice={ originalMonthlyPrice }
+					currencyCode={ currencyCode }
+					original
+					isSmallestUnit
+				/>
+				<PlanPrice rawPrice={ introOfferMonthlyPrice } currencyCode={ currencyCode } />
+			</div>
+			<span className="plan-time-frame">
+				<small>{ billingTimeFrame }</small>
+			</span>
+		</div>
+	);
+};
+
+const preparePlanPriceOfferProps = (
+	selectedPlan: string,
+	plan?: Plan,
+	pricing?: SitePlanPricing
+): PlanPriceOfferProps => {
+	const currencyCode = pricing?.currencyCode;
+	const originalMonthlyPrice = pricing?.originalPrice.monthly || undefined;
+
+	const introOfferRawPrice = pricing?.introOffer?.rawPrice;
+	const introOfferMonthlyPrice = introOfferRawPrice
+		? calculateMonthlyPriceForPlan( selectedPlan, introOfferRawPrice )
+		: undefined;
+
+	const originalFullPrice = pricing?.originalPrice.full || undefined;
+	const introOfferFullPrice = pricing?.introOffer?.rawPrice || undefined;
```

Comment:
I don't think we need `introOfferRawPrice` here. It actually confused me a bit when reading the code.

Also, out of interest, is there a specific reason you're using `|| undefined` rather than `?? undefined`?

```suggestion
	const introOfferFullPrice = pricing?.introOffer?.rawPrice || undefined;
	const introOfferMonthlyPrice = introOfferFullPrice
		? calculateMonthlyPriceForPlan( selectedPlan, introOfferFullPrice )
		: undefined;

	const originalFullPrice = pricing?.originalPrice.full || undefined;	
```